src/features/TransactionStatus/TransactionStatus.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useContext } from 'react';
+import { Input } from '@mycrypto/ui';
+import { withRouter } from 'react-router-dom';
+import isEmpty from 'lodash/isEmpty';
+import queryString from 'query-string';
+
+import { Button, NetworkSelectDropdown, ContentPanel, TxReceipt, InlineMessage } from '@components';
+import { ITxHash, NetworkId, ITxType, ITxConfig, ITxReceipt } from '@types';
+import { NetworkContext, AssetContext, StoreContext, ProviderHandler } from '@services';
+import {
+  makeTxConfigFromTransactionResponse,
+  makePendingTxReceipt,
+  makeTxConfigFromTxReceipt
+} from '@utils/transaction';
+import { noOp } from '@utils';
+import { useEffectOnce, useUpdateEffect } from '@vendor';
+import { DEFAULT_NETWORK, ROUTE_PATHS } from '@config';
+import { translateRaw } from '@translations';
+import { getTxsFromAccount } from '@services/Store/helpers';
+
+const SUPPORTED_NETWORKS: NetworkId[] = ['Ethereum', 'Ropsten', 'Goerli', 'Kovan', 'ETC'];
+
+const TransactionStatus = withRouter(({ history, match, location }) => {
+  const qs = queryString.parse(location.search);
+
+  const { assets } = useContext(AssetContext);
+  const { getNetworkById, networks } = useContext(NetworkContext);
+  const { accounts } = useContext(StoreContext);
+
+  const defaultTxHash = match.params.txHash ? match.params.txHash : '';
+  const defaultNetwork =
+    qs.network && SUPPORTED_NETWORKS.includes(qs.network) ? qs.network : DEFAULT_NETWORK;
+
+  const [txHash, setTxHash] = useState(defaultTxHash);
+  const [networkId, setNetwork] = useState<NetworkId>(defaultNetwork);
+  const [tx, setTx] = useState<{ config: ITxConfig; receipt: ITxReceipt } | undefined>(undefined);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
```

Comment:
Seems like a good time to introduce a reducer.
It will make it a lot easier to test.